### PR TITLE
Fix unset preview points not starting at 40%

### LINKED ISSUE #3111: Preview point is incorrect for maps with "PreviewTime: -1"
The preview point is incorrect for maps having "PreviewTime: -1" set in the .osu file; the music starts playing from the beginning in such cases instead of from 40% into the song.
Example in stable:
https://streamable.com/5q4ok
Same map in lazer:
https://streamable.com/0vr5r

### PR DESCRIPTION
**Pull request by Joehuu**
Fixes #3111.

---

### CODE REVIEW

File: osu.Game/Screens/Multi/Multiplayer.cs [hunk 1]
```diff
@@ -280,11 +280,14 @@ private void updateTrack(ValueChangedEvent<WorkingBeatmap> _ = null)
 
                 if (track != null)
                 {
-                    track.RestartPoint = Beatmap.Value.Metadata.PreviewTime;
                     track.Looping = true;
 
                     if (!track.IsRunning)
+                    {
+                        track.Stop();
+                        track.RestartPoint = Beatmap.Value.Metadata.PreviewTime != -1 ? Beatmap.Value.Metadata.PreviewTime : 0.4f * track.Length;
```
Comment (Aergwyn) [inline osu.Game/Screens/Multi/Multiplayer.cs]: Would it be worth to change the getter of `PreviewTime` to contain this logic instead of having it at each occurrence?

File: osu.Game/Screens/Multi/Multiplayer.cs [hunk 2]
```diff
@@ -280,11 +280,14 @@ private void updateTrack(ValueChangedEvent<WorkingBeatmap> _ = null)
 
                 if (track != null)
                 {
-                    track.RestartPoint = Beatmap.Value.Metadata.PreviewTime;
                     track.Looping = true;
 
                     if (!track.IsRunning)
+                    {
+                        track.Stop();
```
Comment (Aergwyn) [inline osu.Game/Screens/Multi/Multiplayer.cs]: Why was this added?

**Comment by Joehuu** [comment]
Doesn't seem to work after doing some tests (i.e. deleting, importing, etc.) or *something*. Closing until I can make this work 100%.
